core/src/main/java/ai/timefold/solver/core/config/util/ConfigUtils.java

```diff
@@ -372,12 +372,19 @@ public static List<Member> getAllMembers(Class<?> baseClass, Class<? extends Ann
                     .filter(field -> field.isAnnotationPresent(annotationClass) && !field.isSynthetic())
                     .sorted(alphabeticMemberComparator);
             var methodStream = Stream.of(clazz.getDeclaredMethods())
-                    .filter(method -> method.isAnnotationPresent(annotationClass) && !method.isSynthetic())
-                    .sorted(alphabeticMemberComparator);
+                    .filter(method -> method.isAnnotationPresent(annotationClass) && !method.isSynthetic());
+
+            for (var implementedInterface : clazz.getInterfaces()) {
+                methodStream = Stream.concat(
+                        methodStream,
+                        Arrays.stream(implementedInterface.getMethods())
+                                .filter(method -> method.isAnnotationPresent(annotationClass) && !method.isSynthetic()));
+            }
+            methodStream = methodStream.sorted(alphabeticMemberComparator);
             memberStream = Stream.concat(memberStream, Stream.concat(fieldStream, methodStream));
             clazz = clazz.getSuperclass();
         }
-        return memberStream.collect(Collectors.toList());
+        return memberStream.distinct().collect(Collectors.toList());
```

Comment:
Wouldn't it be better to first distinct and then sort? Either way, both can be done at this final stream, and not on the intermediate streams.